src/notifications/util.js

```diff
@@ -0,0 +1,23 @@
+async function tryCallbackOrPromise(promise, callback = null) {
+  if (isValidCallback(callback)) {
+    promise.then((data) => {
+      callback(null, data);
+    })
+      .catch(callback);
+  }
+  await promise;
```

Comment:
I will test this later correctly, right now we can't tests this changes, probably you are right, I'll let you know whenever I test this.